daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/DamlLfEncoder.scala

```diff
@@ -93,38 +99,35 @@ private[daml] object DamlLfEncoder extends App {
       inputFiles: List[String],
       outputFile: String,
       languageVersion: LanguageVersion,
+      validation: Boolean,
   )
 
   private def parseArgs() = {
-    val nAgrs = args.length
-
     @tailrec
-    def go(
-        appArgs: Arguments = Arguments(List.empty, "", LanguageVersion.default),
-        i: Int = 0,
-    ): Arguments =
-      if (i == nAgrs) {
-        if (appArgs.outputFile.isEmpty)
-          error("output file not set")
-        if (appArgs.inputFiles.isEmpty)
-          error("no input files set")
-        else
-          appArgs
-      } else
-        args(i) match {
-          case "--target" if i + 1 < nAgrs =>
-            go(appArgs.copy(languageVersion = LanguageVersion.assertFromString(args(i + 1))), i + 2)
-          case "--output" if i + 1 < nAgrs =>
-            go(appArgs.copy(outputFile = args(i + 1)), i + 2)
-          case _ if i + 1 >= nAgrs =>
-            error(
-              s"usage: encoder_binary inputFile1 ... inputFileN --output outputFile [--target version]"
-            )
-          case x =>
-            go(appArgs.copy(inputFiles = x :: appArgs.inputFiles), i + 1)
-        }
-
-    go()
+    def go(appArgs: Arguments, args: List[String]): Arguments =
```

Comment:
I suggest replacing thisy by a scopt parser, it’s harder any longer and much nicer to maintain. But let’s leave this out of this PR.

Reply:
:+1: